addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestBaseHandler.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.nest.handler;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.nest.internal.NestDeviceDataListener;
+import org.openhab.binding.nest.internal.NestUpdateRequest;
+import org.openhab.binding.nest.internal.data.BaseNestDevice;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Optional;
+import java.util.TimeZone;
+
+import static org.eclipse.smarthome.core.thing.Thing.PROPERTY_MODEL_ID;
+
+/**
+ * Deals with the structures on the Nest api, turning them into a thing in openHAB.
+ *
+ * @author David Bennett - initial contribution
+ * @author Martin van Wingerden - Splitted of NestBaseHandler
+ */
+abstract class NestBaseHandler extends BaseThingHandler implements NestDeviceDataListener {
+    NestBaseHandler(@NonNull Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        getNestBridgeHandler().map(b -> b.addDeviceDataListener(NestBaseHandler.this));
+
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, "Waiting for refresh");
+    }
+
+    @Override
+    public void dispose() {
+        getNestBridgeHandler().map(b -> b.removeDeviceDataListener(NestBaseHandler.this));
+    }
+
+    private Optional<NestBridgeHandler> getNestBridgeHandler() {
+        return Optional.ofNullable(getBridge()).map(b -> (NestBridgeHandler) b.getHandler());
+    }
+
+    boolean isNotHandling(BaseNestDevice nestDevice) {
+        String thingDeviceId = getThing().getThingTypeUID().getId();
```

Comment:
All devices currently remain offline because this line should actually be:
`String thingDeviceId = getThing().getUID().getId();`

Thanks for all the hard work you are putting into this binding!

Reply:
Thanks for your feedback, tonight I will do some testing including the corner cases, restarting reconnecting etc.